index.ts

```diff
@@ -32,12 +34,16 @@ class Directory {
   _instantiationStack: string | undefined;
   __broccoliFeatures__: BroccoliFeatures;
 
-  constructor(directoryPath: string, watched?: boolean | string, options?: DirectoryOptions) {
-    if (typeof directoryPath !== 'string') {
-      throw new Error('Expected a path (string), got ' + directoryPath);
+  constructor(_directoryPath: string, watched?: boolean | string, options?: DirectoryOptions) {
+    if (typeof _directoryPath !== 'string') {
+      throw new Error('Expected a path (string), got ' + _directoryPath);
     }
 
-    this._directoryPath = directoryPath;
+    // toNamespacedPath was added in Node 9, this guard can be removed after
+    // dropping support for Node < 10
+    let directoryPath = path.toNamespacedPath ? path.toNamespacedPath(_directoryPath) : _directoryPath;
+    this._directoryPath = path.resolve(directoryPath);
```

Comment:
Note: even after we can drop the guard for `toNamespacedPath` we will still need to do this `path.resolve`. Specifically, on POSIX systems `toNamespacedPath` always returns the input path unmodified, but on Windows `toNamespacePath` normalizes. 😭

Reply:
![D23ZcurWkAA3zRG](https://user-images.githubusercontent.com/166909/58051991-7d151800-7b08-11e9-80ed-3d0b34c82346.jpg)